### PR TITLE
Automatically cleaning up the temporary directores created

### LINKED ISSUE #31: Make sure the temporary directories are cleaned up on exit
We are currently using mkdtemp, which requires us to delete the directory once we are done with it.
We should instead use tempfile.TemporaryDirectory as a context manager, so when we exit the scope the directory and its contents are deleted for us.

### PR DESCRIPTION
**Pull request by Sush0512**
Fixes #31 
Referred https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

### CODE REVIEW

File: crawler.py [hunk 1]
```diff
@@ -191,10 +191,10 @@ def run_all(driver):
 os.environ['GCOV_PREFIX_STRIP'] = str(strip_count)
 
 # Returns absolute path for created temp file
-gcov_dir = tempfile.mkdtemp()
+gcov_dir = tempfile.TemporaryDirectory()
```
Comment (marco-c) [inline crawler.py]: We should use a context manager here, to make sure the directory is always deleted.
You can do something like:
```Python
with tempfile.TemporaryDirectory() as gcov_dir, tempfile.TemporaryDirectory() as jsvm_dir:
    ...
```

**Comment by Sush0512** [comment]
Hi @marco-c 
Thank you for pointing out, I somehow missed it. I have made the necessary changes now.

**Comment by Sush0512** [comment]
Travis log. Should I add a newline at the end of file?
> $ flake8 .
> ./crawler.py:200:20: W292 no newline at end of file
> The command "flake8 ." exited with 1.

**Comment by marco-c** [comment]
Yes.

**Pull request merged**